src/client/Microsoft.Identity.Client/Cache/Items/MsalAccountCacheItem.cs

```diff
@@ -18,22 +18,21 @@ internal MsalAccountCacheItem()
 
         internal MsalAccountCacheItem(
             string preferredCacheEnv,
-            MsalTokenResponse response,
+            string clientInfo,
+            IdToken idToken,
             string preferredUsername,
             string tenantId)
             : this()
         {
-            var idToken = IdToken.Parse(response.IdToken);
```

Comment:
Small optimization - we seem to be parsing the IdToken like crazy. This is pretty expensive - it involves splitting a string and parsing a piece of JSON. Not a big deal, but when done n times ...

Reply:
Something which should be done as optimizations everywhere then? string operations seems to be one of the operations which causes the most CPU and memory usage...

---
In reply to: [401011046](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1735#discussion_r401011046) [](ancestors = 401011046)

Reply:
Yes, and now I've done the same for ClientInfo, which is also a small piece of json that we were parsing like crazy.